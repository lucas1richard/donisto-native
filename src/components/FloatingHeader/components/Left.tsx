import * as React from 'react';
import Touchable from 'components/Touchable';
import { View, StyleSheet } from 'react-native';
import { Feather } from '@expo/vector-icons';
import Txt from 'components/Txt';

const hitSlop = {
  top: 15,
  bottom: 15,
  left: 50,
  right: 50
};

function Left({ noBackOption, onBack }: {
  noBackOption?: boolean,
  blue?: boolean,
  onBack: () => void
}) {
  if (noBackOption) {
    return <View style={styles.wrapper} />;
  }

  return (
    <Touchable onPress={onBack} hitSlop={hitSlop}>
      <View style={styles.wrapper}>
        <Feather
          size={30}
          name="chevron-left"
          color="#fff"
        />
        <Txt color="light">Back</Txt>
      </View>
    </Touchable>
  );
}

export default Left;

const styles = StyleSheet.create({
  wrapper: {
    position: 'absolute',
    flexDirection: 'row',
    alignItems: 'center',
    top: 12,
    left: 12,
    backgroundColor: 'rgba(0,0,0,0.2)',
    padding: 2,
    paddingRight: 10,
    paddingBottom: 4,
    borderRadius: 99
  }
});
