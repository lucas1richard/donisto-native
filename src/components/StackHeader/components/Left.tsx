import * as React from 'react';
import Touchable from 'components/Touchable';
import { View, StyleSheet } from 'react-native';
import { Feather } from '@expo/vector-icons';
import theme from 'theme/iftheme';

const hitSlop = {
  top: 15,
  bottom: 15,
  left: 50,
  right: 50
};

export interface ILeftProps {
  noBackOption?: boolean,
  onBack: () => void
}

function Left({ noBackOption, onBack }: ILeftProps) {
  if (noBackOption) {
    return <View style={styles.wrapper} />;
  }

  return (
    <Touchable onPress={onBack} hitSlop={hitSlop}>
      <View style={styles.wrapper}>
        <Feather
          size={30}
          name="chevron-left"
          color={theme.primary[500]}
        />
      </View>
    </Touchable>
  );
}

export default Left;

const styles = StyleSheet.create({
  wrapper: {
    // padding: 2,
    // paddingRight: 4,
    // paddingBottom: 4,
    // borderRadius: 20
    height: 34,
    width: 34,
    padding: 2,
    borderRadius: 17,
    zIndex: 100
  }
});
