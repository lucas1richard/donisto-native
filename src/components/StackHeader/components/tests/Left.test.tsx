import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Left, { ILeftProps } from '../Left';

describe('<Left />', () => {
  const renderComponent = (props: ILeftProps) => toJson(shallow(<Left {...props} />));
  it('renders', () => {
    expect(renderComponent({
      onBack: jest.fn()
    })).toMatchSnapshot();
  });
});