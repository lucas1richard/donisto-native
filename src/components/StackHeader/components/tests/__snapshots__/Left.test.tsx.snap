// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Left /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Left
    onBack={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <Component
        style={
          Object {
            "borderRadius": 17,
            "height": 34,
            "padding": 2,
            "width": 34,
            "zIndex": 100,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="rgb(251, 100, 65)"
          name="chevron-left"
          size={30}
        />
      </Component>,
      "hitSlop": Object {
        "bottom": 15,
        "left": 50,
        "right": 50,
        "top": 15,
      },
      "onPress": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Icon
          allowFontScaling={false}
          color="rgb(251, 100, 65)"
          name="chevron-left"
          size={30}
        />,
        "style": Object {
          "borderRadius": 17,
          "height": 34,
          "padding": 2,
          "width": 34,
          "zIndex": 100,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": false,
          "color": "rgb(251, 100, 65)",
          "name": "chevron-left",
          "size": 30,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Component
          style={
            Object {
              "borderRadius": 17,
              "height": 34,
              "padding": 2,
              "width": 34,
              "zIndex": 100,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="rgb(251, 100, 65)"
            name="chevron-left"
            size={30}
          />
        </Component>,
        "hitSlop": Object {
          "bottom": 15,
          "left": 50,
          "right": 50,
          "top": 15,
        },
        "onPress": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Icon
            allowFontScaling={false}
            color="rgb(251, 100, 65)"
            name="chevron-left"
            size={30}
          />,
          "style": Object {
            "borderRadius": 17,
            "height": 34,
            "padding": 2,
            "width": 34,
            "zIndex": 100,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": false,
            "color": "rgb(251, 100, 65)",
            "name": "chevron-left",
            "size": 30,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
